### PR TITLE
lib: python.ts add exception type

### PR DESCRIPTION
**Pull request by tomasmatus**
Add exception type to handle cases when spawned python script exits with error.

### CODE REVIEW

**Comment by tomasmatus** [comment]
I'm not really sure if the `exit_status` can also be null as I would expect any shell to have an exist status but it's probably better to be safe and also assume it can be null.

**Comment by tomasmatus** [comment]
ok it can be

https://github.com/cockpit-project/cockpit/blob/9601aee69e8606957fcd17091d60107cedfdf576/pkg/lib/cockpit.js#L1399

**Review by jelly — COMMENTED** [review]
Forgot to publish

File: pkg/lib/python.ts [hunk 1]
```diff
@@ -17,10 +17,15 @@
  * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
  */
 
-import cockpit from "cockpit";
+import cockpit, { BasicError } from "cockpit";
 
 const pyinvoke = ["sh", "-ec", "exec $(command -v /usr/libexec/platform-python || command -v python3) -c \"$@\"", "--"];
 
+export interface PythonExitStatus extends BasicError {
+    exit_status: number,
```
[the remark this answers is not in the document]
Comment (tomasmatus) [reply pkg/lib/python.ts]: Indeed, I have a feeling that it's a bug and instead it should be checking for `this.exit_status !== null`

File: pkg/lib/cockpit.js [hunk 2]
```diff
@@ -1407,7 +1407,7 @@ function factory() {
                 this.message = cockpit.message(options.problem);
             else if (this.exit_signal !== null)
                 this.message = cockpit.format(_("$0 killed with signal $1"), name, this.exit_signal);
-            else if (this.exit_status !== undefined)
+            else if (this.exit_status !== null)
```
Comment (cockpituous) [inline pkg/lib/cockpit.js:1410]: This added line is not executed by any test.

**Review by jelly — APPROVED** [review]

**Pull request merged**